modules/tracer-otel/src/main/java/org/opensearch/tracing/OTelSpan.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Default implementation of {@link Span} using Otel span. It keeps a reference of OpenTelemetry Span and handles span
+ * lifecycle management by delegating calls to it.
+ */
+class OTelSpan extends AbstractSpan {
+
+    private static final Logger logger = LogManager.getLogger(OTelSpan.class);
+
+    private final io.opentelemetry.api.trace.Span otelSpan;
+
+    public OTelSpan(String spanName, io.opentelemetry.api.trace.Span span, Span parentSpan, Level level) {
+        super(spanName, parentSpan, level);
+        this.otelSpan = span;
+        logger.trace(
```

Comment:
These log statements are not needed, if we need to report traces in logs we should provide the `SpanProcessor` implementation for that.

Reply:
Wouldn't adding span processor just to log on span start/end events be an overkill?

Reply:
But this is opt-in - one needs it - one adds it (vs logs, I am trying to imagine under which circumstances these logging statements are useful)

Reply:
Done